color/src/main/java/com/afollestad/materialdialogs/color/DialogColorChooserExt.kt

```diff
@@ -72,3 +80,164 @@ fun MaterialDialog.colorChooser(
+}
+
+private fun onCustomValueChanged(
+  dialog: MaterialDialog,
+  supportCustomAlpha: Boolean,
+  waitForPositiveButton: Boolean,
+  valueChanged: Boolean,
+  customView: View,
+  vColor: View,
+  sbAlpha: SeekBar, sbRed: SeekBar, sbGreen: SeekBar, sbBlue: SeekBar,
+  tvAlphaValue: TextView, tvRedValue: TextView, tvGreenValue: TextView, tvBlueValue: TextView,
+  selection: ColorCallback) {
+  if (supportCustomAlpha) {
+    tvAlphaValue.text = sbAlpha.progress.toString()
+  }
+  tvRedValue.text = sbRed.progress.toString()
+  tvGreenValue.text = sbGreen.progress.toString()
+  tvBlueValue.text = sbBlue.progress.toString()
+
+  val color = Color.argb(if (supportCustomAlpha) sbAlpha.progress else 255, sbRed.progress, sbGreen.progress, sbBlue.progress)
+  vColor.setBackgroundColor(color)
+  // simple solution - we save the color as view tag
+  if (valueChanged) {
+    customView.setTag(color)
+    dialog.setActionButtonEnabled(POSITIVE, true)
+  }
+
+  if (!waitForPositiveButton && valueChanged) {
+    selection?.invoke(dialog, color)
+  }
+}
+
+// ----------------
+// Helper functions
+// ----------------
+
+private fun selectedColor(dialog: MaterialDialog, allowCustomColor: Boolean) : Int? {
+  if (allowCustomColor) {
+    val viewPager = getPager(dialog)
+    if (viewPager.currentItem == 1) {
+      return getPageCustomView(dialog).getTag() as Int?
+    }
+  }
+  return (getPageGridView(dialog).adapter as ColorGridAdapter).selectedColor()
+}
+
+private fun getPageGridView(dialog: MaterialDialog): RecyclerView {
+  return dialog.findViewById(R.id.rvGrid) as RecyclerView
+}
+
+private fun getPageCustomView(dialog: MaterialDialog): View {
+  return dialog.findViewById(R.id.llCustomColor)
+}
+
+private fun getPager(dialog: MaterialDialog): ViewPager {
+  return dialog.findViewById(R.id.vpPager)
+}
+
+private fun getTabLayout(dialog: MaterialDialog): TabLayout {
+  return dialog.findViewById(R.id.tlTabls)
+}
+
+private fun setLayoutPager(dialog: MaterialDialog, allowCustomColor: Boolean, @StringRes resTabGrid: Int?, textTabGrid: String?, @StringRes resTabCustom: Int?, textTabCustom: String?) {
+  dialog.customView(R.layout.md_color_chooser_pager)
+  val viewPager = getPager(dialog)
+  viewPager.adapter = ColorPagerAdapter(dialog, resTabGrid, textTabGrid, resTabCustom, textTabCustom)
+  viewPager.addOnPageChangeListener(object :ViewPager.OnPageChangeListener {
+    override fun onPageScrollStateChanged(state: Int) {}
+
+    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
+
+    override fun onPageSelected(position: Int) {
+      dialog.setActionButtonEnabled(POSITIVE, selectedColor(dialog, allowCustomColor) != null)
+    }
+  })
+  val tabLayout = getTabLayout(dialog)
+  tabLayout.setupWithViewPager(viewPager)
+}
+
+private fun setLayoutColorPresets(dialog: MaterialDialog) {
+  dialog.customView(R.layout.md_color_chooser_grid)
+}
+
+private fun tintSeekbar(seekBar: SeekBar, color: Int) {
```

Comment:
Might be nice to make this an extension function on `SeekBar`

Reply:
done as well